core/diagnostic.go

```diff
@@ -30,3 +34,127 @@ func (c *Core) DumpTokenChain(dr *model.TCDumpRequest) *model.TCDumpReply {
 	ds.NextBlockID = nextID
 	return ds
 }
+
+func (c *Core) DumpSmartContractTokenChain(dr *model.TCDumpRequest) *model.TCDumpReply {
```

Comment:
Again , feel we can generalise This to DumpTokenChain() .

Reply:
Yes i think this can be clubbed together